mit-cml/appinventor-sources
# Extensions upgrader

Allowing and Handling Extensions Upgrades. Gracefully.

How to test this : 
Download all the [Test Aix Files](https://drive.google.com/open?id=0B6TjsdbpeaqcWWdudVV6a01jRzQ) and upgrade a component by the same method as importing a component.
Upgrade the aix files in any order. 
You shouldn't receive any errors.
You shouldn't lose your blocks.
You shouldn't lose your properties panel values.
You should be happy.


If you do encounter any problem, please do tell me. 
If you have better idea on how to handle a particular situation more gracefully please do tell.
You can also create your own weird aix upgrade and test. 
Test without mercy. :)
